TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -634,18 +634,15 @@ public override void Exit(Node node)
         /// <summary>
         /// Link this sentence to the current section or paragraph if any.
         /// </summary>
-        /// <param name="sentence">The sentence to link.</param>
+        /// <param name="sentence">The sentence to link.</param>       
         private void LinkBlockSentenceToCurrentSectionParagraph(Sentence sentence)
         {
-            var currentProcedure = (Procedure) this.CurrentProgramCfgBuilder.CurrentParagraph ??
-                                   this.CurrentProgramCfgBuilder.CurrentSection;
-
-            if (currentProcedure != null)
+            if (sentence.Procedure != null)
```

Comment:
Now that I read it again, `sentence.Procedure` should never be null since we have a default root section.

Reply:
Yep, there is a bug in presence of DECLARATIVE:
https://github.com/TypeCobolTeam/TypeCobol/issues/2081